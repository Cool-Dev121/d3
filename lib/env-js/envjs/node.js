require('./platform/node');
Envjs.eventLoop = function() {}; // disabled for clean shutdown

require('./window');
new Window(__this__); // initialize a blank window
