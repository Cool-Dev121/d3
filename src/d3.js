import "start";
import "core/core";

// Compatibility
import "compat/date";
import "compat/style";

// Arrays
import "core/ascending";
import "core/descending";
import "core/min";
import "core/max";
import "core/extent";
import "core/sum";
import "core/mean";
import "core/median";
import "core/quantile";
import "core/bisect";
import "core/shuffle";
import "core/permute";
import "core/zip";
import "core/transpose";
import "core/keys";
import "core/values";
import "core/entries";
import "core/merge";
import "core/range";
import "core/nest";
import "core/map";
import "core/set";

// Math
import "core/random";

// String Formatting
import "core/requote";
import "core/round";

// Loading External Resources
import "core/xhr";
import "core/text";
import "core/json";
import "core/html";
import "core/xml";

// Namespaces
import "core/ns";

// Internals
import "core/functor";
import "core/rebind";
import "core/dispatch";

///////////




import "core/format";
import "core/formatPrefix";
import "core/ease";
import "core/event";
import "core/transform";
import "core/interpolate";
import "core/uninterpolate";
import "core/color";
import "core/rgb";
import "core/hsl";
import "core/hcl";
import "core/lab";
import "core/xyz";
import "core/selection";
import "core/selection-select";
import "core/selection-selectAll";
import "core/selection-attr";
import "core/selection-classed";
import "core/selection-style";
import "core/selection-property";
import "core/selection-text";
import "core/selection-html";
import "core/selection-append";
import "core/selection-insert";
import "core/selection-remove";
import "core/selection-data";
import "core/selection-datum";
import "core/selection-filter";
import "core/selection-order";
import "core/selection-sort";
import "core/selection-on";
import "core/selection-each";
import "core/selection-call";
import "core/selection-empty";
import "core/selection-node";
import "core/selection-transition";
import "core/selection-root";
import "core/selection-enter";
import "core/selection-enter-select";
import "core/transition";
import "core/transition-select";
import "core/transition-selectAll";
import "core/transition-filter";
import "core/transition-attr";
import "core/transition-style";
import "core/transition-text";
import "core/transition-remove";
import "core/transition-ease";
import "core/transition-delay";
import "core/transition-duration";
import "core/transition-each";
import "core/transition-transition";
import "core/transition-tween";
import "core/timer";
import "core/mouse";
import "core/touches";
import "core/noop";

import "scale/scale";
import "scale/nice";
import "scale/linear";
import "scale/bilinear";
import "scale/polylinear";
import "scale/log";
import "scale/pow";
import "scale/sqrt";
import "scale/ordinal";
import "scale/category";
import "scale/quantile";
import "scale/quantize";
import "scale/threshold";
import "scale/identity";

import "svg/svg";
import "svg/arc";
import "svg/line";
import "svg/line-radial";
import "svg/area";
import "svg/area-radial";
import "svg/chord";
import "svg/diagonal";
import "svg/diagonal-radial";
import "svg/symbol";
import "svg/axis";
import "svg/brush";

import "behavior/behavior";
import "behavior/drag";
import "behavior/zoom";

import "layout/layout";
import "layout/bundle";
import "layout/chord";
import "layout/force";
import "layout/partition";
import "layout/pie";
import "layout/stack";
import "layout/histogram";
import "layout/hierarchy";
import "layout/pack";
import "layout/cluster";
import "layout/tree";
import "layout/treemap";
import "layout/voronoi";

import "geo/geo";
import "geo/area";
import "geo/bounds";
import "geo/cartesian";
import "geo/centroid";
import "geo/circle";
import "geo/clip";
import "geo/clip-antimeridian";
import "geo/clip-circle";
import "geo/clip-polygon";
import "geo/clip-view";
import "geo/compose";
import "geo/distance";
import "geo/graticule";
import "geo/haversin";
import "geo/greatArc";
import "geo/interpolate";
import "geo/length";
import "geo/path-area";
import "geo/path-buffer";
import "geo/path-centroid";
import "geo/path-context";
import "geo/path";
import "geo/projection";
import "geo/resample";
import "geo/rotation";
import "geo/spherical";
import "geo/stream";
import "geo/albers";
import "geo/albers-usa";
import "geo/azimuthal";
import "geo/azimuthal-equal-area";
import "geo/azimuthal-equidistant";
import "geo/conic";
import "geo/conic-conformal";
import "geo/conic-equal-area";
import "geo/conic-equidistant";
import "geo/equirectangular";
import "geo/gnomonic";
import "geo/mercator";
import "geo/orthographic";
import "geo/stereographic";
import "geo/transverse-mercator";

import "dsv/dsv";
import "dsv/csv";
import "dsv/tsv";

import "time/time";
import "time/format-localized";
import "time/format";
import "time/format-utc";
import "time/format-iso";
import "time/interval";
import "time/second";
import "time/minute";
import "time/hour";
import "time/day";
import "time/week";
import "time/month";
import "time/year";
import "time/scale";
import "time/scale-utc";

import "geom/geom";
import "geom/hull";
import "geom/polygon";
import "geom/voronoi";
import "geom/delaunay";
import "geom/quadtree";

import "end";
