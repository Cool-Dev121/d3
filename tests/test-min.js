require("./../lib/env-js/envjs/node");
require("./../d3");

console.log("min:");
console.log("  " + d3.min([1, 2, 3, 4, 5]));
console.log("");

console.log("min with accessor function:");
console.log("  " + d3.min([[1, 2, 3, 4, 5], [2, 4, 6, 8, 10]], function(d) {
  return d3.max(d);
}));
console.log("");

console.log("min index:");
console.log("  " + d3.min([1, 2, 3, 4, 5], function(d, i) {
  return i;
}));
console.log("");

console.log("min with first element NaN:");
console.log("  " + d3.min([NaN, 1, 2, 3, 4, 5]));
console.log("");

console.log("min with last element NaN:");
console.log("  " + d3.min([1, 2, 3, 4, 5, NaN]));
console.log("");

console.log("min with null and undefined elements:");
console.log("  " + d3.min([10, null, 3, undefined, 5, NaN]));
console.log("");
