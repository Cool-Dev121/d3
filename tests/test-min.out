min:
  1

min with accessor function:
  5

min index:
  0

min with first element NaN:
  1

min with last element NaN:
  1

min with null and undefined elements:
  3

